R/input-utils.R

```diff
@@ -90,41 +90,77 @@ generateOptions <- function(inputId, selected, inline, type = 'checkbox',
   div(class = "shiny-options-group", options)
 }
 
-# Take a vector or list, and convert to named list. Also, if any children are
-# vectors with length > 1, convert those to list. If the list is unnamed,
-# convert it to a named list with blank names.
-listify <- function(x) {
-  if (is.list(x) || is.null(x)) {
-    asNamedVector(lapply(x, listify))
-  } else if (is.character(x)) {
-    if (length(x) == 1 && is.null(names(x))) {
-      x
-    } else {
-      as.list(asNamedVector(x))
-    }
-  } else {
-    # Can get here if x is a factor.
-    listify(stats::setNames(as.character(x), names(x)))
-  }
+# True when a choice list item represents a group of related inputs.
+isGroup <- function(choice) {
+  length(choice) > 1 || !is.null(names(choice))
```

Comment:
Is this condition correct for a leaf node where it's a named list/vector with an empty name?

```R
x <- list(A="a", "b")[2]
x
#> [[1]]
#> [1] "b"
dput(x)
structure(list("b"), .Names = "")
```

Reply:
In the existing code it's considered a group, so the condition is correct in that respect. I can see a good argument for changing that behavior, though. That object is leafier than it is groupy.